### DISCUSSION TITLE
Fehlerhaftes CHARGING_COMPLETED mit Keba Wallbox

### DISCUSSION THREAD

**Discussion by SpeGo23** [camueller/SmartApplianceEnabler#603] [Installation und Betrieb]
Bei mir läuft SAE nun schon eine Zeit lang in Verbindung mit einer Keba P30 Wallbox, aber leider lief es noch nie stabil. Oft helfen Neustarts von SAE oder der Wallbox und heute bin ich endlich mal dazu gekommen, im Log nachzusehen was eigentlich schief läuft. Heute war ein "guter Tag", d.h. Überschussladen hat die meiste Zeit des Tages sehr gut funktioniert und wenn ich richtig schaue, fingen die Probleme erst gegen 16 Uhr an. Im Log sehe ich, dass SAE "CHARGING_COMPLETED" feststellt, obwohl das Auto erst bei gut 40% steht. Um 15:05 hatte ich mal die Konfig leicht geändert um einen SAE Neustart zu erzwingen, das war aber keine echte inhaltliche Änderung. Danach gab es noch mehr Probleme (z.B. SAE ist "grün" obwohl tatsächlich nicht geladen wird) bis hin zu einer nicht mehr verfügbaren Keba App, aber ich würde mal mit dem ersten Fehler anfangen, vielleicht gab es danach eine Kette von Fehlern. Woran kann es liegen, dass CHARGING_COMPLETED fehlerhaft festgestellt wird? Eigentlich gab es dann nochmal viel Sonne und die Wallbox hätte laden sollen. Danke fürs Anschauen, das Log von heute hänge ich an!
[Uploading rolling-2024-05-05v5_FehlerAbends_25mb.log…]()

**Comment by camueller** [comment]
Das Anhängen des Logs hat wohl nicht geklappt - insofern kann ich nichts konkretes sagen.

Wenn der SAE fälschlicherweise CHARGING_COMPLETED erkennt, kann das mehrere Ursachen haben:
- der aktuelle SOC (eingegeben oder von SOC-Script geliefert) zu Beginn des Ladens war falsch
- die Batteriekapazität ist falsch konfiguriert
- die Wallbox reagiert auf den Einschaltbefehl nicht

Die Keba-Wallboxen mögen es gar nicht, wenn Modbus-Zugriffe parallel erfolgen, d.h. wenn ausser dem SAE noch andere Programme oder Apps genutzt werden.

**Comment by SpeGo23** [comment]
[rolling-2024-05-05v5_FehlerAbends_25mb.log](https://github.com/camueller/SmartApplianceEnabler/files/15233387/rolling-2024-05-05v5_FehlerAbends_25mb.log)
Ich versuche nun nochmal das Log hochzuladen, die möglichen Ursachen helfen mir leider noch nicht so richtig weiter. Den SOC Wert geben wir noch nicht an bzw. es läuft noch kein Skript und die Batterkapazität sollte korrekt konfiguriert sein. Wir lassen mittlerweile auch bewusst nur den SAE laufen und bedienen die Keba App nicht, um Parallelzugriffe zu verhindern. Kannst du aus dem Log etwas herauslesen?

**Comment by camueller** [comment]
In Deinem Log sind 37 Fehler enthalten, die beim Lesen von Registern aufgetreten sind, z.B.
```
2024-05-05 09:02:39,687 TRACE [Timer-0] d.a.s.m.e.ReadHoldingRegisterExecutor [ReadHoldingRegisterExecutor.java:38] F-23051988-000000000001-00: Reading holding register=1000 requestWords=1
2024-05-05 09:02:42,806 ERROR [Timer-0] d.a.s.m.EVModbusControl [EVModbusControl.java:185] F-23051988-000000000001-00: Error reading register 1000
com.ghgande.j2mod.modbus.ModbusIOException: Executing transaction 00 19 00 00 00 06 FF 03 03 E8 00 01 failed (tried 5 times) General exception - failed to read - Connection reset
```
Wenn die Kommunikation dermassen gestört ist, kann man kein korrektes Verhalten der Software (also des SAE) erwarten.

Im Rahmen des Supports verschiedener User hat sich gezeigt, dass die Modbus-Implementierung von Keba nicht wirklich stabil läuft. Wenn ich mich richtig erinnere, hatten die Keba-Wallbox auch anfangs gar keine Modbus-Unterstützung, die wurde dann irgendwann "nachgereicht". Leider wurde wohl nicht genug getestet ...
